rclc/include/rclc/init.h

```diff
@@ -104,6 +104,112 @@ rcl_ret_t
 rclc_support_fini(
   rclc_support_t * support);
 
+/**
+ *  Allocates the rclc_support_t object on the heap.
+ *  Can be used as if no stack allocation can or should be used.
+ *
+ *  * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | No
+ *
+ * \return pointer to the support (rclc_support_t)
+ * \return NULL, if no memory could be allocated.
+ */
+RCLC_PUBLIC
+rclc_support_t *
+rclc_support_alloc();
```

Comment:
`rclc_support_init` already takes the allocator in the argument, i think `rclc_support_init` should return `rclc_support_t *` when succeeds? in that case, we can use allocator specified by the user application and store it in the `rclc_support_t *` to deallocate on `rclc_support_fini`?

Reply:
Sorry, I didn't get this, probably because I have only a vague model in my mind about rcl_allocator_t.

I get that `rclc_support_init` already takes the allocator in the argument. But why should `rclc_support_init` return `rclc_support_t *`? `rclc_support_init` also takes `support` in the argument as an inout parameter, i.e. it already implicitly returns `rclc_support_t *`.
Or are you referring to `rclc_support_alloc`? Should this already take the allocator to allocate the memory?

Additionaly, `rclc_support_init_with_options` (which is called from `rclc_support_init`) already stores the provided allocator in `rclc_support_t *`. 
Or are you referring to `rclc_support_free`? Should this make use of the allocator stored in `rclc_support_t` to free the memory? That would indeed make sense if `rclc_support_alloc` used the allocator to allocate the memory.

Reply:
I think this is now addressed with the recent change. If not, please point out what is missing exactly.